core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -585,4 +586,32 @@ public String errorMessage() {
 
     }
 
+    static class DefaultClusterNameCheck implements Check {
+
+        private final ClusterName clusterName;
+
+        DefaultClusterNameCheck(final ClusterName clusterName) {
+            this.clusterName = clusterName;
+        }
+
+        @Override
+        public boolean check() {
+            return clusterName.equals(ClusterName.DEFAULT);
```

Comment:
So, this is me thinking aloud, but would it be better to ensure that there is a /settings/ for `cluster.name` rather than checking if it's the default cluster name? What if someone explicitly had:

```
cluster.name: elasticsearch
```

In their configuration, and wanted it to be named "elasticsearch"?